qtp_target_gene/validate.py

```diff
@@ -193,44 +193,51 @@ def _validate_per_sample_FASTQ(qclient, job_id, prep_info, files):
                      % (samples_count, read_files_count, rev_count))
         return False, None, error_msg
 
-    if 'run_prefix' in prep_info[samples[0]]:
-        # The column 'run_prefix' is present in the prep information.
-        # Make sure that twe have the same number of run_prefix values
-        # than the number of samples
-        run_prefixes = [v['run_prefix'] for k, v in prep_info.items()]
-        if samples_count != len(set(run_prefixes)):
-            repeated = ["%s (%d)" % (p, run_prefixes.count(p))
-                        for p in set(run_prefixes)
-                        if run_prefixes.count(p) > 1]
-            error_msg = ("The values for the column 'run_prefix' are not "
-                         "unique for each sample. Repeated values: %s"
-                         % ', '.join(repeated))
-            return False, None, error_msg
-
-        error_msg = ("The provided files do not match the run prefix values "
-                     "in the prep information. Offending files: "
-                     "raw_forward_seqs: %s, raw_reverse_seqs: %s")
-    else:
-        # The column 'run_prefix' is not in the prep template. In this case,
-        # check that the files are prefixed by the sample ids without the
-        # study id
-        run_prefixes = [sid.split('.', 1)[1] for sid in samples]
-        error_msg = ("The provided files are not prefixed by sample id. "
-                     "Please provide the 'run_prefix' column in your prep "
-                     "information. Offending files: raw_forward_seqs: %s, "
-                     "raw_reverse_seqs: %s")
-
-    # Check that the provided files match the run prefixes
-    fwd_fail = [basename(fp) for fp in read_files
-                if not basename(fp).startswith(tuple(run_prefixes))]
-    if rev_count > 0:
-        rev_fail = [basename(fp) for fp in files['raw_reverse_seqs']
+    def _check_files(run_prefixes, read_files, rev_count, files):
```

Comment:
Is this indentation correct? I don't think I've seen private functions used like this within other functions.